src/imageio/imageio_avif.h

```diff
@@ -21,6 +21,20 @@
 #include "common/image.h"
 #include "common/mipmap_cache.h"
 
+#if AVIF_VERSION > 110100
+  #define DT_avifImageSetProfileICC avifImageSetProfileICC
+  #define DT_avifRGBImageAllocatePixels avifRGBImageAllocatePixels
+  #define DT_avifImageSetMetadataExif avifImageSetMetadataExif
+  #define DT_avifImageSetMetadataXMP avifImageSetMetadataXMP
+  #define DT_avifImageRGBToYUV avifImageRGBToYUV
+#else
+  #define DT_avifImageSetProfileICC AVIF_RESULT_OK; avifImageSetProfileICC
```

Comment:
These macros are quite brittle, e.g. you couldn't use the function call as part of the condition in an `if` statement.  I would have made them inline functions.

Reply:
We now have three pr's for this issue. I think it's better to have fixes either via macros or inlines than spread over the code. You wanna take over?